source/ada/lsp-ada_documents.adb

```diff
@@ -492,6 +523,7 @@ package body LSP.Ada_Documents is
 
          Add (Old_Line_Number);
          Free (LCS);
+         Sanitize;
```

Comment:
Instead of Sanitize I would add the following code:

         if Self.Line_To_Index.Find
           (Integer (Edit.Last_Element.span.last.line)) = No_Element
         then
            declare
               Element : LSP.Messages.TextEdit := Edit.Last_Element;
            begin
               Element.span.last :=
                 (line => Line_Number
                    (Old_Lines.Last_Index - Old_Lines.First_Index),
                  character =>
                    UTF_16_Index (Length (Old_Lines.Last_Element)));
               Edit.Replace_Element (Edit.Last, Element);
            end;
         end if;

In this case we will not raise an exception and also we don't need to process all elements because we can have only one TextEdit for the last line in the end of the list

Reply:
I like the idea